git-branching-merging.qmd

```diff
@@ -81,6 +81,24 @@ Because of this, we have disabled merge commits in our repositories.
 Instead, we rely on either 'rebase and merge' (preferred option) if commits are grouped logically and have clear error messages, or 'squash and merge' if we don't want to keep the commit history (as a last resort; if the commit history is too messy and impossible to clean, or if there is a merge commit in the history).
 If necessary and in order to make a 'rebase' rather than a 'squash', the PR history can be rewritten & cleaned using a command-line interactive git rebase, and squashing, re-ordering, rewording, etc. as necessary and as described in [this blog post](https://github.blog/2022-06-30-write-better-commits-build-better-projects/).
 
+## Who merges the pull request?
+
+By convention, the package maintainer is in charge of merging the pull requests.
+This is indeed often more convenient as they are aware of other ongoing activities in the package that may impact or be impacted with the incoming changes.
+We also have a guarantee that maintainers have sufficient permissions to merge the pull request, and potentially bypass some checks if necessary, which may not be the case of the contributor.
+
+This recommendation remains valid even if the maintainer if the author of the pull request and they request review from a non-maintainer.
+
+As a contributor, if you want to signal your changes are not ready to be merged, you should mark your pull requests as draft.
```

Comment:
I like that this is mentioned. Do we also want to consider explicitly tagging a reviewer before reviewing can begin or can people just review if they feel the need to? I think the former introduces a little bit of friction but solves the issue where a PR seems to be open for review but is not ready or is still being actively worked on. The other solution would be to only open a PR for review after it was previously set to draft.